### PR TITLE
CLI: Re-Add Nuxt support

### PR DESCRIPTION
**Pull request by valentinpalkovic**
## What I did

Reverts storybookjs/storybook#28479

#### The changes in this PR are covered in the following automated tests:
- [ ] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

### Documentation

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

This pull request has been released as version `0.0.0-pr-28607-sha-54ec63fe`. Try it out in a new sandbox by running `npx storybook@0.0.0-pr-28607-sha-54ec63fe sandbox` or in an existing project with `npx storybook@0.0.0-pr-28607-sha-54ec63fe upgrade`.
<details>
<summary>More information</summary>

| | |
| --- | --- |
| **Published version** | [`0.0.0-pr-28607-sha-54ec63fe`](https://npmjs.com/package/storybook/v/0.0.0-pr-28607-sha-54ec63fe) |
| **Triggered by** | @shilman |
| **Repository** | [storybookjs/storybook](https://github.com/storybookjs/storybook) |
| **Branch** | [`revert-28479-revert/26884`](https://github.com/storybookjs/storybook/tree/revert-28479-revert/26884) |
| **Commit** | [`54ec63fe`](https://github.com/storybookjs/storybook/commit/54ec63fe3b92e8368c9f286acb1cedb23e5b6923) |
| **Datetime** | Sat Dec 14 16:19:12 UTC 2024 (`1734193152`) |
| **Workflow run** | [12331349549](https://github.com/storybookjs/storybook/actions/runs/12331349549) |

To request a new release of this pull request, mention the `@storybookjs/core` team.

_core team members can create a new canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=28607`_
</details>

| name | before | after | diff | z | % |
| ---- | ------ | ----- | ---- | - | - |
| createSize |  0 B | 0 B | 0 B | - | - |
| generateSize |  77.7 MB | 77.7 MB | 0 B | **4.32** | 0% |
| initSize |  136 MB | 136 MB | 819 B | **4.36** | 0% |
| diffSize |  58.4 MB | 58.4 MB | 819 B | **4.36** | 0% |
| buildSize |  7.24 MB | 7.24 MB | 0 B | **2.38** | 0% |
| buildSbAddonsSize |  1.88 MB | 1.88 MB | 0 B | **2.38** | 0% |
| buildSbCommonSize |  195 kB | 195 kB | 0 B | - | 0% |
| buildSbManagerSize |  1.86 MB | 1.86 MB | 0 B | **2.01** | 0% |
| buildSbPreviewSize |  0 B | 0 B | 0 B | - | - |
| buildStaticSize |  0 B | 0 B | 0 B | - | - |
| buildPrebuildSize |  3.93 MB | 3.93 MB | 0 B | **2.38** | 0% |
| buildPreviewSize |  3.3 MB | 3.3 MB | 0 B | **2.25** | 0% |
| testBuildSize |  0 B | 0 B | 0 B | - | - |
| testBuildSbAddonsSize |  0 B | 0 B | 0 B | - | - |
| testBuildSbCommonSize |  0 B | 0 B | 0 B | - | - |
| testBuildSbManagerSize |  0 B | 0 B | 0 B | - | - |
| testBuildSbPreviewSize |  0 B | 0 B | 0 B | - | - |
| testBuildStaticSize |  0 B | 0 B | 0 B | - | - |
| testBuildPrebuildSize |  0 B | 0 B | 0 B | - | - |
| testBuildPreviewSize |  0 B | 0 B | 0 B | - | - |

| name | before | after | diff | z | % |
| ---- | ------ | ----- | ---- | - | - |
| createTime |  24s | 23.8s | -209ms | **1.43** | -0.9% |
| generateTime |  19.7s | 18.5s | -1s -276ms | **-1.36** | 🔰-6.9% |
| initTime |  14.2s | 12.5s | -1s -719ms | -1.23 | -13.7% |
| buildTime |  14.2s | 8.8s | -5s -422ms | -0.35 | -61.2% |
| testBuildTime |  0ms | 0ms | 0ms | - | - |
| devPreviewResponsive |  5s | 4.3s | -722ms | **-1.5** | 🔰-16.7% |
| devManagerResponsive |  3.6s | 3.2s | -419ms | **-1.49** | 🔰-12.9% |
| devManagerHeaderVisible |  596ms | 513ms | -83ms | -0.85 | -16.2% |
| devManagerIndexVisible |  627ms | 541ms | -86ms | **-1.25** | 🔰-15.9% |
| devStoryVisibleUncached |  1.4s | 1.7s | 251ms | -0.2 | 14.4% |
| devStoryVisible |  625ms | 583ms | -42ms | -0.74 | -7.2% |
| devAutodocsVisible |  554ms | 446ms | -108ms | -1.04 | -24.2% |
| devMDXVisible |  517ms | 447ms | -70ms | -0.9 | -15.7% |
| buildManagerHeaderVisible |  689ms | 510ms | -179ms | -0.97 | -35.1% |
| buildManagerIndexVisible |  805ms | 599ms | -206ms | -0.99 | -34.4% |
| buildStoryVisible |  631ms | 474ms | -157ms | -0.86 | -33.1% |
| buildAutodocsVisible |  446ms | 400ms | -46ms | -0.7 | -11.5% |
| buildMDXVisible |  479ms | 439ms | -40ms | -0.25 | -9.1% |

## Greptile Summary

This PR reverts the removal of Nuxt support in Storybook, restoring functionality that was previously removed due to QA issues before the 8.2 release.

- Added Nuxt framework detection and configuration in `detect.ts` and `project_types.ts`, using `@storybook-vue/nuxt` package
- Added Nuxt generator with specific configuration in `generators/NUXT/index.ts`, including `@nuxtjs/storybook` module integration
- Added Nuxt sandbox template `nuxt-vite/default-ts` marked as `inDevelopment` with Vite builder
- Added special case handling for Nuxt in Yarn configuration to avoid log filtering
- Added TypeScript detection for Nuxt projects via tsconfig.json even without direct TypeScript dependency

The changes appear comprehensive but lack test coverage and documentation updates that should be addressed before merging.

### CODE REVIEW

**Comment by valentinpalkovic** [comment]
@tobiasdiez 
I am encountering the following error after init while testing the revert:

1. npx nuxi@latest init (npm)
2. cd ./nuxt-app
3. npx storybook@0.0.0-pr-28607-sha-b9bf7d39 init

```
Running Storybook

> storybook
> storybook dev -p 6006 --quiet

storybook v0.0.0-pr-28607-sha-b9bf7d39

WARN The following packages are incompatible with Storybook 0.0.0-pr-28607-sha-b9bf7d39 as they depend on different major versions of Storybook packages:
WARN - @nuxtjs/storybook@8.1.5
WARN  Repo: https://storybook.nuxtjs.org
WARN 
WARN 
WARN Please consider updating your packages or contacting the maintainers for compatibility details.
WARN For more on Storybook 8 compatibility, see the linked GitHub issue:
WARN https://github.com/storybookjs/storybook/issues/26031
The CJS build of Vite's Node API is deprecated. See https://vitejs.dev/guide/troubleshooting.html#vite-cjs-node-api-deprecated for more details.
Error: Cannot find module '@storybook/builder-vite'
Require stack:
- ./node_modules/@storybook/core/dist/core-server/index.cjs
- ./node_modules/storybook/dist/generate.cjs
- ./node_modules/storybook/bin/index.cjs
    at Module._resolveFilename (node:internal/modules/cjs/loader:1140:15)
    at Module._resolveFilename (./node_modules/esbuild-register/dist/node.js:4799:36)
    at Function.resolve (node:internal/modules/helpers:188:19)
    at getPreviewBuilder (./node_modules/@storybook/core/dist/core-server/index.cjs:102320:19)
    at buildDevStandalone (./node_modules/@storybook/core/dist/core-server/index.cjs:104759:5)
    at async withTelemetry (./node_modules/@storybook/core/dist/core-server/index.cjs:103500:12)
    at async dev (./node_modules/storybook/dist/generate.cjs:735:506)
    at async Command.<anonymous> (./node_modules/storybook/dist/generate.cjs:737:245)

WARN Broken build, fix the error above.
WARN You may need to refresh the browser.
```

**Comment by valentinpalkovic** [comment]
@tobiasdiez
I am encountering the browser following error after init while testing the revert:

npx nuxi@latest init (pnpm)
cd ./nuxt-app
npx storybook@0.0.0-pr-28607-sha-b9bf7d39 init
Running Storybook

```
SyntaxError: The requested module '/node_modules/.pnpm/lodash@4.17.21/node_modules/lodash/kebabCase.js?v=726791be' does not provide an export named 'default'
```

@ndelangen This might be CPC related, though.

**Comment by tobiasdiez** [comment]
Thanks for this PR and your tests!

> Error: Cannot find module '@storybook/builder-vite'
> Require stack:
> - ./node_modules/@storybook/core/dist/core-server/index.cjs
> - ./node_modules/storybook/dist/generate.cjs
> - ./node_modules/storybook/bin/index.cjs
>     at Module._resolveFilename (node:internal/modules/cjs/loader:1140:15)
>     at Module._resolveFilename (./node_modules/esbuild-register/dist/node.js:4799:36)
>     at Function.resolve (node:internal/modules/helpers:188:19)
>     at getPreviewBuilder (./node_modules/@storybook/core/dist/core-server/index.cjs:102320:19)
>     at buildDevStandalone (./node_modules/@storybook/core/dist/core-server/index.cjs:104759:5)
>     at async withTelemetry (./node_modules/@storybook/core/dist/core-server/index.cjs:103500:12)
>     at async dev (./node_modules/storybook/dist/generate.cjs:735:506)
>     at async Command.<anonymous> (./node_modules/storybook/dist/generate.cjs:737:245)

I can reproduce it as well. The problem is that in our setup `@storybook/builder-vite` is only a dependency of the nuxt module, but `require` is called from the main storybook (or core), and thus only looks for it in the main `node_modules` folder. 

Simplest solution: just add `@storybook/builder-vite` as a main dependency. But then the user has to sync the builder-vite version with what the nuxt module is using (or we declare it as peer-dep). Would work but maybe not ideal

More involved solution: make it possible that presets return a javascript object as builder, instead of only a module name. (Similar to how eslint v9 is handling plugins now). 

Do you have any preference, or another idea?

**Comment by tobiasdiez** [comment]
> @tobiasdiez I am encountering the browser following error after init while testing the revert:
> 
> npx nuxi@latest init (pnpm) cd ./nuxt-app npx storybook@0.0.0-pr-28607-sha-b9bf7d39 init Running Storybook
> 
> ```
> SyntaxError: The requested module '/node_modules/.pnpm/lodash@4.17.21/node_modules/lodash/kebabCase.js?v=726791be' does not provide an export named 'default'
> ```
> 
> @ndelangen This might be CPC related, though.

I've fixed this already in the main branch by adding lodash to optimizeDeps:
https://github.com/nuxt-modules/storybook/blob/64e80dec7c15d9e03fa22f806ac269fbe8146d1f/packages/storybook-addon/src/preset.ts#L171-L175

**Comment by valentinpalkovic** [comment]
> Thanks for this PR and your tests!
> 
> > Error: Cannot find module '@storybook/builder-vite'
> > Require stack:
> > 
> > * ./node_modules/@storybook/core/dist/core-server/index.cjs
> > * ./node_modules/storybook/dist/generate.cjs
> > * ./node_modules/storybook/bin/index.cjs
> >   at Module._resolveFilename (node:internal/modules/cjs/loader:1140:15)
> >   at Module._resolveFilename (./node_modules/esbuild-register/dist/node.js:4799:36)
> >   at Function.resolve (node:internal/modules/helpers:188:19)
> >   at getPreviewBuilder (./node_modules/@storybook/core/dist/core-server/index.cjs:102320:19)
> >   at buildDevStandalone (./node_modules/@storybook/core/dist/core-server/index.cjs:104759:5)
> >   at async withTelemetry (./node_modules/@storybook/core/dist/core-server/index.cjs:103500:12)
> >   at async dev (./node_modules/storybook/dist/generate.cjs:735:506)
> >   at async Command. (./node_modules/storybook/dist/generate.cjs:737:245)
> 
> I can reproduce it as well. The problem is that in our setup `@storybook/builder-vite` is only a dependency of the nuxt module, but `require` is called from the main storybook (or core), and thus only looks for it in the main `node_modules` folder.
> 
> Simplest solution: just add `@storybook/builder-vite` as a main dependency. But then the user has to sync the builder-vite version with what the nuxt module is using (or we declare it as peer-dep). Would work but maybe not ideal
> 
> More involved solution: make it possible that presets return a javascript object as builder, instead of only a module name. (Similar to how eslint v9 is handling plugins now).
> 
> Do you have any preference, or another idea?

@tobiasdiez Take a look at https://github.com/storybookjs/storybook/blob/next/code/frameworks/angular/src/preset.ts#L31. You can see that you can also just pass an absolute path as a value. With this approach, you make sure that `@storybook/builder-vite` or any other package which you have installed as a `dependency` will be resolved from `node_modules/@storybook-vue/nuxt` instead of `node_modules/@storybook/core-preset`

**Comment by tobiasdiez** [comment]
Many thanks. This worked like a charm! 🚀 

In https://github.com/nuxt-modules/storybook/pull/729, I've now also added e2e tests (using this PR here to add storybook to a nuxt starter). Running `build-storybook` now works (and is tested), but I still get some errors when actually running the storybook. Is there a reusable test suite or something like this that I can point to a `storybook-static` folder and that verifies that the stories are okay. In a similar vain, it would be handy to have the storybook templates used for the tests here somehow published so that they can be reused to test downstream packages (I noticed this need not only for the nuxt integration but also while developing the vue addon https://github.com/tobiasdiez/storybook-vue-addon).

**Comment by kasperpeulen** [comment]
@valentinpalkovic @tobiasdiez Maybe it is best if we have Nuxt sandbox over here:
https://github.com/storybookjs/storybook/blob/a1f56b3339153f3038a932c84e3374a90db6dd68/code/lib/cli/src/sandbox-templates.ts#L354

**Comment by valentinpalkovic** [comment]
@tobiasdiez As Kasper mentioned, we could add Nuxt to our sandboxes, and you could test the sandbox on this PR and rerun it every time you release a new Canary version. Otherwise, I don't see an easy way to copy over our test suite setup to your repository.

**Comment by KeremTurkyilmaz** [comment]
Hello @tobiasdiez do you have any news when this draft will be merged ?

**Comment by tobiasdiez** [comment]
@kasperpeulen @valentinpalkovic Thanks for the suggestion. Adding a sandbox would indeed give some confidence that things are working, but I was hoping for a solution that we can apply in downstream projects (e.g. that PRs to the nuxt storybook addon are not breaking things). Similar to https://www.npmjs.com/package/@apollo/server-integration-testsuite.

@KeremTurkyilmaz We first need to fix the major bugs in the nuxt module. Help moving this forward is of course very welcome!

**Comment by valentinpalkovic** [comment]
@tobiasdiez Unfortunately, we don't have a dedicated testsuite solution like Apollo does. :/

**Comment by tobiasdiez** [comment]
I've added a nuxt sandbox, but not sure if I did it correctly (it appears it is not run as part of the ci)

**Comment by tobiasdiez** [comment]
We do have default stories in https://github.com/nuxt-modules/storybook/tree/main/packages/storybook-addon/template/cli, but I have no idea how they get installed nor why this would fail in the sandbox. If you give me some pointers where to start, I can look into this.

**Comment by tobiasdiez** [comment]
Can we please get this in as soon as possible? We constantly have people opening issues at the nuxt-storybook module because they configured something wrong. Having the cli integration would make setting this up with nuxt way easier and robust. 🙇

**Comment by tobiasdiez** [comment]
@storybookjs/core, we would greatly appreciate support from the core team regarding the Nuxt-Storybook integration. @chakAs3 and I are working on this in our free time, and unfortunately, we don't have the resources to continually address new issues that arise in the Nuxt module due to upstream changes in Storybook. Many of these issues could be avoided if the Nuxt integration sandbox were included in your test suite.

Additionally, our installation instructions are more complex than necessary because the Storybook CLI still doesn't fully recognize Nuxt projects. Thus new users make mistakes during the setup, and then open issues at the nuxt repo adding to our workload. It would be incredibly helpful if we could get this PR merged as soon as possible. 

Could you also review my pending PRs at your earliest convenience? (https://github.com/storybookjs/storybook/pulls/tobiasdiez) If you believe my approach in a PR isn't the right solution (e.g., [PR #29172 comment](https://github.com/storybookjs/storybook/pull/29172#issuecomment-2396180201) or [PR #28595](https://github.com/storybookjs/storybook/pull/28595)), that’s completely understandable. However, it would be immensely helpful if you would then implement an alternative solution (or at least suggest one). These PRs are not just for theoretical improvements; they address critical issues that make maintaining the Nuxt integration extremely challenging.

Thank you for your time and consideration!

**Comment by valentinpalkovic** [comment]
I have informed the core team and we will try to prioritise this and the mentioned PRs asap.

**Comment by chakAs3** [comment]
Yes, this has become quite challenging for users. Just last week, I helped a company manually set up Storybook on a large Nuxt project. I had previously built a custom CLI for this, but stopped maintaining it due to the difficulty of keeping it in sync with Storybook's frequent releases.

@tobiasdiez, on a related note, I’m encountering consistent failures when building and publishing Storybook to Chromatic, likely due to the large number of components and insufficient RAM. Have you come across this issue, and is there a recommended solution?

**Comment by valentinpalkovic** [comment]
I talked to the team, and currently, we are wrapping up the 8.4 release. Next week, I will get the chance to take a proper look and hopefully get the PR merged.

**Comment by valentinpalkovic** [comment]
Finally taking a look...

**Comment by tobiasdiez** [comment]
> Our CI is finally green. 

Awesome! Thanks for you work!

> When generating a sandbox, it does not contain our default stories (Button, Header, Page), which some of our e2e-production tests require. Therefore, I was forced to disable our e2e-production tests on the nuxt sandbox. If you add the [default stories](https://github.com/storybookjs/storybook/tree/next/code/renderers/vue3/template/cli) to https://github.com/nuxt-modules/storybook/tree/main/packages/storybook-addon/template/cli, we may be able to reactivate the e2e-production tests.

I'm not so sure if that's a good idea. Sure, for e2e tests it makes a lot of sense to have these standard stories. However, as a user of the nuxt module these auto-generated files are mostly noise and I think most users would immediately delete them.

Is it possible that you use the vue-package to generate the stories for the e2e tests here in this repo?

(By the way, it would perhaps be an improvement for the on-boarding if instead of auto-generating Buttons and similar controls that are not really useful, these would be provided by the storybook package - similarly, how nuxt provides the welcome screen that we show in the storybook. Then storybook init would only generate one or two stories that show users how everything works.)

**Comment by tobiasdiez** [comment]
Sorry, I wrote that in a bit of a hurry.

Currently, if you run `storybook init` it generates a Button and a story for that button. From a users perspective, the button component is totally useless - the story could at least in principle be reused/modified to serve as a first real story of the actual components in the app. My idea was to distribute the mock button as part of the storybook framework, instead of automatically generating it. The story would then look like
```js
import { fn } from '@storybook/test';
import type { Meta, StoryObj } from '@storybook/vue3';

---- only change here

// Change the following to import the actual component, e.g.  './Button.vue'
import Button from '@storybook/vue3/mocks';

----- end of change

// More on how to set up stories at: https://storybook.js.org/docs/writing-stories
const meta = {
  title: 'Example/Button',
  component: Button,
  // This component will have an automatically generated docsPage entry: https://storybook.js.org/docs/writing-docs/autodocs
  tags: ['autodocs'],
  argTypes: {
    size: { control: 'select', options: ['small', 'medium', 'large'] },
    backgroundColor: { control: 'color' },
  },
  args: {
    primary: false,
    // Use `fn` to spy on the onClick arg, which will appear in the actions panel once invoked: https://storybook.js.org/docs/essentials/actions#action-args
    onClick: fn(),
  },
} satisfies Meta<typeof Button>;

export default meta;
type Story = StoryObj<typeof meta>;
/*
 *👇 Render functions are a framework specific feature to allow you control on how the component renders.
 * See https://storybook.js.org/docs/api/csf
 * to learn how to use render functions.
 */
export const Primary: Story = {
  args: {
    primary: true,
    label: 'Button',
  },
};

export const Secondary: Story = {
  args: {
    primary: false,
    label: 'Button',
  },
};

export const Large: Story = {
  args: {
    label: 'Button',
    size: 'large',
  },
};

export const Small: Story = {
  args: {
    label: 'Button',
    size: 'small',
  },
};
```

**Comment by shilman** [comment]
Non-starter. We did that originally and docgen doesn't work well unless we have the source code.

Next year I'd like to transition to an onboarding experience where the user provides their own components instead of these throwaways. However, we'll need to do a bunch of work to make that experience smooth for users.

**Comment by hacknug** [comment]
> > When generating a sandbox, it does not contain our default stories (Button, Header, Page), which some of our e2e-production tests require. Therefore, I was forced to disable our e2e-production tests on the nuxt sandbox. If you add the [default stories](https://github.com/storybookjs/storybook/tree/next/code/renderers/vue3/template/cli) to https://github.com/nuxt-modules/storybook/tree/main/packages/storybook-addon/template/cli, we may be able to reactivate the e2e-production tests.
> 
> I'm not so sure if that's a good idea. Sure, for e2e tests it makes a lot of sense to have these standard stories. However, as a user of the nuxt module these auto-generated files are mostly noise and I think most users would immediately delete them.

Can't we just have a flag to generate those files so users don't ever see them and the sandbox can use them for testing? 

It would _only_ require making changes to `@nuxtjs/storybook` and adding another item to the options passed to `packageManager.runPackageCommand` in `code/lib/create-storybook/src/generators/NUXT/index.ts`.

Am I missing something here?

Also, is that the only thing blocking this PR from being merged?

**Comment by hacknug** [comment]
> It would _only_ require making changes to `@nuxtjs/storybook` and adding another item to the options passed to `packageManager.runPackageCommand` in `code/lib/create-storybook/src/generators/NUXT/index.ts`.

This assumption was not correct as those are the arguments passed to `nuxi` (aka nuxt's cli).

From what I can see, the change should be made to [`storybook-nuxt`](https://github.com/storybook-vue/storybook-nuxt/blob/main/packages/storybook-nuxt-cli/src/index.ts) since that is [what the module docs' recommend for initializing a project](https://storybook.nuxtjs.org/getting-started/setup#storybook-config-files). Not sure if this step was part of the CI implementation though.

I'd say adding `-f, --full-init` or `-f, --full` might be a good way to ensure the cli alone can generate the files required for CI to pass.

Only issue I see is that the repo description for [`storybook-nuxt`](https://github.com/storybook-vue/storybook-nuxt) states the project was deprecated in favor of [`nuxt-modules/storybook`](https://github.com/nuxt-modules/storybook/) so perhaps it should be migrated to the module repo 😄

File: code/lib/create-storybook/src/generators/baseGenerator.ts [hunk 1]
```diff
@@ -169,8 +170,18 @@ const stripVersions = (addons: string[]) => addons.map((addon) => getPackageDeta
...
+const hasFrameworkTemplates = (framework?: SupportedFrameworks) => {
+  if (!framework) {
+    return false;
+  }
+  // Nuxt has framework templates, but for sandboxes we create them from the Vue3 renderer
+  // As the Nuxt framework templates are not compatible with the stories we need for CI.
+  // See: https://github.com/storybookjs/storybook/pull/28607#issuecomment-2467903327
+  if (framework === 'nuxt') {
+    return process.env.IN_STORYBOOK_SANDBOX !== 'true';
+  }
```
Comment (kasperpeulen) [inline code/lib/create-storybook/src/generators/baseGenerator.ts:182]: @tobiasdiez @hacknug I now only create the default stories (using those from the vue3 renderer) in the sandboxes.

Comment (tobiasdiez) [reply code/lib/create-storybook/src/generators/baseGenerator.ts:182]: Sounds good to me!

Ideally, you would test against https://github.com/nuxt-modules/storybook/tree/main/examples/showcase, which has the highest coverage of features.

**Comment by kasperpeulen** [comment]
@tobiasdiez 
I can only get the Button, Header, Page stories to work if I add this rule:
```
  viteFinal: (config) => mergeConfig(config,{
    resolve: {
      alias: {
        "vue": "vue/dist/vue.esm-bundler.js",
      }
    },
```
to `main.ts`

Otherwise I get this warning:
> Component provided template option but runtime compilation is not supported in this build of Vue. Configure your bundler to alias "vue" to "vue/dist/vue.esm-bundler.js". 

I saw that other people got the same warning:
https://github.com/nuxt-modules/storybook/issues/808
https://github.com/nuxt-modules/storybook/issues/805#issuecomment-2428578367

We do add this automatically in the vue3-vite framework:
https://github.com/storybookjs/storybook/blob/a70b0693147d92838bd3645b4137f41ca8108d9f/code/frameworks/vue3-vite/src/plugins/vue-template.ts#L3-L14

Maybe this PR would solve this?
https://github.com/nuxt-modules/storybook/pull/740

**Comment by chakAs3** [comment]
> I can only get the Button, Header, Page stories to work if I add this rule:
> 
> ```
>   viteFinal: (config) => mergeConfig(config,{
>     resolve: {
>       alias: {
>         "vue": "vue/dist/vue.esm-bundler.js",
>       }
>     },
> ```

These stories use render functions with string templates, which means they need to be compiled at runtime.

**Comment by tobiasdiez** [comment]
> Maybe this PR would solve this? [nuxt-modules/storybook#740](https://github.com/nuxt-modules/storybook/pull/740)

Yes, this is the proper way to fix it. But for this to work, we first need https://github.com/storybookjs/storybook/pull/29795.

How should we coordinate the new releases? Can you temporarily add the special vite config for nuxt in the e2e tests? Then once https://github.com/storybookjs/storybook/pull/29795 is released, we merge https://github.com/nuxt-modules/storybook/pull/740 and cut a new release?

**Comment by yannbf** [comment]
> > Maybe this PR would solve this? [nuxt-modules/storybook#740](https://github.com/nuxt-modules/storybook/pull/740)
> 
> Yes, this is the proper way to fix it. But for this to work, we first need #29795.
> 
> How should we coordinate the new releases? Can you temporarily add the special vite config for nuxt in the e2e tests? Then once #29795 is released, we merge [nuxt-modules/storybook#740](https://github.com/nuxt-modules/storybook/pull/740) and cut a new release? Afterwards the workaround can be removed again here in the main repo.

Awesome! I merged #29795 and will be making a release soon. Also added the e2e workaround for now! 🤞

**Comment by yannbf** [comment]
@tobiasdiez the fix you wanted is now part of [v8.4.7](https://github.com/storybookjs/storybook/releases/tag/v8.4.7) (as well as 8.5.0-alpha.18)! I merged the latest `next` into this branch, hopefully CI is all good now 🤞

**Comment by kasperpeulen** [comment]
CI is allmost green, one failing E2E test, trying to get this merged this week.

**Comment by tobiasdiez** [comment]
Thanks so much, @kasperpeulen and @yannbf, for shepherding this in!

**Pull request merged**